src/mapping/AxialGeoMultiscaleMapping.cpp

```diff
@@ -58,46 +60,54 @@ void AxialGeoMultiscaleMapping::mapConsistent(DataID inputDataID, DataID outputD
 {
   PRECICE_TRACE(inputDataID, outputDataID);
 
-  const Eigen::VectorXd &inputValues     = input()->data(inputDataID)->values();
-  int                    valueDimensions = input()->data(inputDataID)->getDimensions();
-  Eigen::VectorXd &      outputValues    = output()->data(outputDataID)->values();
+  const Eigen::VectorXd &inputValues        = input()->data(inputDataID)->values();
+  int                    inValueDimensions  = input()->data(inputDataID)->getDimensions();
+  int                    outValueDimensions = output()->data(outputDataID)->getDimensions();
+  Eigen::VectorXd &      outputValues       = output()->data(outputDataID)->values();
+
+  int coord = -1;
+  if (_axis == X) {
+    coord = 0;
+  } else if (_axis == Y) {
+    coord = 1;
+  } else if (_axis == Z) {
+    coord = 2;
+  } else {
+    PRECICE_ASSERT(false, "Unknown axis.");
+  }
 
-  PRECICE_ASSERT(valueDimensions == output()->data(outputDataID)->getDimensions(),
-                 valueDimensions, output()->data(outputDataID)->getDimensions());
-  PRECICE_ASSERT(inputValues.size() / valueDimensions == static_cast<int>(input()->vertices().size()),
-                 inputValues.size(), valueDimensions, input()->vertices().size());
-  PRECICE_ASSERT(outputValues.size() / valueDimensions == static_cast<int>(output()->vertices().size()),
-                 outputValues.size(), valueDimensions, output()->vertices().size());
+  PRECICE_ASSERT((inputValues.size() / inValueDimensions == static_cast<int>(input()->vertices().size())),
+                 inputValues.size(), inValueDimensions, input()->vertices().size());
+  PRECICE_ASSERT((outputValues.size() / outValueDimensions == static_cast<int>(output()->vertices().size())),
+                 outputValues.size(), outValueDimensions, output()->vertices().size());
 
   PRECICE_DEBUG("Map consistent");
   if (_type == SPREAD) {
-    PRECICE_ASSERT(inputValues.size() == valueDimensions);
+    PRECICE_ASSERT(inputValues.size() == 1);
```

Comment:
A base class would indeed be a good idea. We could do that in a follow-up (please open an issue), unless you think that it would already simplify the radial case.

Reply:
Remember to transfer the issues from https://github.com/ezonta/precice/issues to this repository and add them to the project board: https://github.com/orgs/precice/projects/14 (I see that it was set to private, changed it to public).

Reply:
Issues are added, but I think I cannot add them to the project board.

Reply:
Added them, thanks!